crawl-ref/source/items.cc

```diff
@@ -129,6 +129,8 @@ static bool _merge_items_into_inv(item_def &it, int quant_got,
 static bool will_autopickup   = false;
 static bool will_autoinscribe = false;
 
+static int last_auto_pickup_try = 1;
+
```

Comment:
I'm not happy about this adding ui-influencing global state, but I don't know where something like this would go otherwise. @aidanholm can you comment here?

Reply:
Yeah, this is not a good idea; the better approach is to do this logic in `main.cc` and check `prev_cmd` there.

Reply:
I was wondering whether there was something like that in the code base, but couldn't find it.
Do you also have a suggestion oh how to avoid triggering pickup when one has just closed a different window with ; ?

Reply:
Sorry, not sure I understand the question; `crawl_state.prev_cmd` should only be set to `CMD_INSPECT_FLOOR` if that was the previous command run, so you shouldn't be able to get to a state where you closed some other window with `;` without `crawl_state.prev_cmd` being set to something else.

Reply:
I misunderstood the way prev_cmd worked. You're right, the bug I described can't occur. I have now updated the logic.